python/lsst/skymap/packers.py

```diff
@@ -31,10 +69,46 @@ class SkyMapDimensionPacker(DimensionPacker):
     Parameters
     ----------
     fixed : `lsst.daf.butler.DataCoordinate`
-        Expanded data ID that must include at least the skymap dimension.
-    dimensions : `lsst.daf.butler.DimensionGraph`
+        Data ID that identifies just the ``skymap`` dimension.  Must have
+        dimension records attached unless ``n_tracts`` and ``n_patches`` are
+        not `None`.
+    dimensions : `lsst.daf.butler.DimensionGraph`, optional
         The dimensions of data IDs packed by this instance.  Must include
-        skymap, tract, and patch, and may include band.
+        ``{skymap, tract, patch}``, and may include ``band``.  If not provided,
+        this will be set to include ``band`` if if ``n_bands != 0``.
+    bands : `~collections.abc.Mapping` [ `str`, `int` ] or `None`, optional
+        Mapping from band name to integer to use in the packed ID.  `None` uses
+        a fixed set of bands defined in this class, but this should be avoided
+        in new code.
```

Comment:
I'm nervous about default values that "should be avoided in new code": better to default to `{}` or something, so that people aren't surprised by getting the hard-coded list by default?

Reply:
As with the config field, I'm walking back the admonition here instead, as I think the default is useful even if an explicit value is better.